src/util.ts

```diff
@@ -79,6 +79,16 @@ export function isTruthy<T>(item: T | false | null | undefined): item is T {
   return Boolean(item);
 }
 
+
+/** Get the package name from a given import specifier (ex: "preact" from "preact/hooks") */
+export function getPackageNameFromSpecifier(specifier: string): string {
+  let [packageName, ...deepPackagePathParts] = specifier.split('/');
+  if (packageName.startsWith('@')) {
+    packageName += '/' + deepPackagePathParts.shift();
+  }
+  return packageName;
+}
```

Comment:
What if this function returned a `{ packageName, deepImport }` object containing the rest of the specifier (or a `[string, string]` tuple)? I can see that being a generally-useful thing where we either use the entry points if the `deepImport` doesn’t exist, or we try and resolve if it does.

Reply:
🤔 I guess it’d also be good to centralize the export map logic somewhere, too, but maybe that’d be overloading it here.

Reply:
+1, we actually have this function exactly as you've described it already in the CDN repo. I'll port that into Snowpack to match